mindvalley/wukong-cli
# PXP-594: Find a solution to support a canary api in the CLI

## Summary


This is the starting point of supporting **canary** API.


Ticket: [PXP-594](https://mindvalley.atlassian.net/browse/PXP-594)

## What's Changed



### Added
- [x] Add a `canary` flag (e.g. `wukong pipeline list --canary`)

### Changed
- [x] canary channel will include `mv-canary-stage: Always` in the request header (for some reason, the `reqwest` will make the header name lowercase)



## How to test
1. run wukong command with `--canary` flag and `--vvv` flag, e.g. `cargo run -- pipeline list --vvv --canary`
2. check the header value, you should see the `mv-canary-stage: Always` header
<img width="685" alt="Screenshot 2023-08-28 at 5 07 26 PM" src="https://github.com/mindvalley/wukong-cli/assets/7545747/bebcea67-7b02-474e-afbb-b7417899a786">



[PXP-594]: https://mindvalley.atlassian.net/browse/PXP-594?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
